pkgs/applications/version-management/gogs/default.nix

```diff
@@ -0,0 +1,445 @@
+# This file was generated by go2nix.
+{ stdenv, lib, makeWrapper, go, goPackages, git, fetchgit, sqliteSupport ? true }:
+
+with goPackages;
+
+buildGoPackage rec {
+  name = "gogs-${version}";
+  version = "20160227-${stdenv.lib.strings.substring 0 7 rev}";
+  rev = "d320915ad2a7b4bbab075b98890aa50f91f0ced5";
+
+  buildInputs = [ makeWrapper go ];
+  buildFlags = lib.optional (sqliteSupport) "-tags sqlite";
+  goPackagePath = "github.com/gogits/gogs";
+
+  postInstall = ''
+    wrapProgram $bin/bin/gogs \
+      --prefix PATH : ${git}/bin \
+      --run 'export GOGS_WORK_DIR=${"$" + "{"}GOGS_WORK_DIR:-$PWD}' \
```

Comment:
Use the nix string open / close symbol to escape, ex ''${}
On Sun, Feb 28, 2016 at 8:25 PM Valérian Galliat <notifications@github.com>
wrote:

> In pkgs/applications/version-management/gogs/default.nix
> <https://github.com/NixOS/nixpkgs/pull/13566#discussion_r54368220>:
>
> > +
> > +with goPackages;
> > +
> > +buildGoPackage rec {
> > +  name = "gogs-${version}";
> > +  version = "20160227-${stdenv.lib.strings.substring 0 7 rev}";
> > +  rev = "d320915ad2a7b4bbab075b98890aa50f91f0ced5";
> > +
> > +  buildInputs = [ makeWrapper go ];
> > +  buildFlags = lib.optional (sqliteSupport) "-tags sqlite";
> > +  goPackagePath = "github.com/gogits/gogs";
> > +
> > +  postInstall = ''
> > +    wrapProgram $bin/bin/gogs \
> > +      --prefix PATH : ${git}/bin \
> > +      --run 'export GOGS_WORK_DIR=${"$" + "{"}GOGS_WORK_DIR:-$PWD}' \
>
> This line looks odd, because I didn't find any way to escape ${ inside a
> Nix string.
>
> The documentation <http://nixos.org/nix/manual/#idm46912467695696> says
> to put a \ in front of it but this did not work for me, still yielding unexpected
> $undefined error.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/13566/files#r54368220>.
>